cmd/crdvalidator/handlers/crd.go

```diff
@@ -73,3 +77,11 @@ func (cv *CrdValidator) InjectDecoder(d *admission.Decoder) error {
 	cv.decoder = d
 	return nil
 }
+
+// enabled takes a CRD and checks its content to see crdvalidator
+// is disabled explicitly
+func enabled(crd *apiextensionsv1.CustomResourceDefinition) bool {
+	// Check if the annotation to disable validation is set
+	return crd.GetAnnotations()[annotation.ValidationKey] == annotation.Enabled ||
+		crd.GetAnnotations()[annotation.ValidationKey] == ""
```

Comment:
I think if we're thinking of this CRD validator as a standalone thing, it may actually make more sense to require opting into it.

If we think opt-in is "bad" for a specific rukpak scenario, we can always have a provisioner implementation inject this annotation automatically or based on a `BundleInstance` annotation.

Reply:
So we had an implementation that did just that but it got some push back to be opt-out, specifically for RukPak. If we feel it is a good idea I don't mind going back to the opt-in mechanic for the entire cluster but having RukPak inject the annotation. I think there was also some concern around the fact that we're adding annotations to CRD's that did not previously exist in the bundle prior to resolving.

Reply:
Yeah it seems like the feedback was that it's less useful in non-rukpak contexts. And that's actually why I changed my mind and think it needs to be opt-in. Imagine if a cluster admin installed this webhook because they wanted rukpak. That would mean that all existing and new CRDs would need to consider this annotation, regardless of whether that CRD was/is managed by rukpak.

Reply:
Ah, so actually there is a new change that this webhook _only_ receives CRDs owned by a BundleInstance. Accordingly, regular CRD's on the cluster will not be validated against this webhook. The only time you need to worry about this annotation is if you want to disable CRD validation for a CRD within a RukPak Bundle. 

[See this change for reference](https://github.com/operator-framework/rukpak/blob/e38a5a36ff9f3d0ecc637fa1c9d433b8d4989f93/manifests/crdvalidator/04_webhook.yaml#L21-L23).

Reply:
Ok, that solves that concern then. In that case, I think I'm good with this merging.

I think I would still like to revisit the opt-in/opt-out question (even regardless of it _just_ applying to rukpak) as its own thing, mainly just to make sure we have a focused discussion about _just_ that without also thinking about the validation itself, which I think we all agree is pretty useful.

Reply:
Maybe this is something worth bubbling up as a topic in the upstream OLM WG meeting later this week?